Analysis/include/AnalysisCategories.h

```diff
@@ -35,14 +36,16 @@ ENUM_NAMES(DYFitModel) = {
 
 struct EventRegion {
     static const EventRegion& Unknown();
-    static const EventRegion& OS_Isolated();
-    static const EventRegion& OS_AntiIsolated();
-    static const EventRegion& SS_Isolated();
-    static const EventRegion& SS_LooseIsolated();
-    static const EventRegion& SS_AntiIsolated();
-    static const EventRegion& SignalRegion();
+    static const EventRegion& OS_Isolated(SignalMode& signal_mode);
+    static const EventRegion& OS_AntiIsolated(SignalMode& signal_mode);
+    static const EventRegion& SS_Isolated(SignalMode& signal_mode);
+    static const EventRegion& SS_LooseIsolated(SignalMode& signal_mode);
+    static const EventRegion& SS_AntiIsolated(SignalMode& signal_mode);
+    static const EventRegion& SignalRegion(SignalMode& signal_mode);
+    static const SignalObjectSelector InitializeWP(SignalMode& signal_mode);
```

Comment:
the interface of `OS_Isolated` and other methods shouldn't change. Keep them as they are.
InitializeWP should be `static void Initialize(DiscriminatorWP iso_lower, DiscriminatrWP anti_iso_lower, DiscriminatorWP anti_iso_upper)`.

in cpp implementation of Initialize should be something like:
```
_OS_Isolated = EventRegion().SetCharge(true).SetLowerIso(iso_lower);
_OS_AntiIsolated = EventRegion().SetCharge(true).SetLowerIso(anti_iso_lower).SetUpperIso(anti_iso_upper);
...
```
where `_OS_Isolated` etc. are defined as a static member of EventRegion class:
```
static boost::optional<EventRegion> _OS_Isolated;
```
And implementation of method `OS_Isolated` etc. should be like:
```
if(!_OS_Isolated. is_initialized())
    throw exception("Event regions are not initialized");
return *_OS_Isolated;
```